src/main/kotlin/net/davoleo/javajuice/math/vector/Vector2d.kt

```diff
@@ -12,10 +16,23 @@ class Vector2d(a: Double, b: Double) : Vector {
         get() = 2
 
     override fun norm(type: Int): Double {
-        TODO("Not yet implemented")
+        // Norma 1
+        return when (type) {
+            1 -> vec[0] + vec[1]
+            // Norma 2
+            2 -> sqrt(vec[0].pow(2.0) + vec[1].pow(2.0))
+            // Norma inf
+            else -> max(vec[0], vec[1])
+        }
```

Comment:
Penso sia meglio utilizzare una costante unica per la norma infinito, e generalizzare le altre norme in modo che si possa fare una norma con qualsiasi numero, altrimenti qualcuno che magari vuole fare una norma 3 si ritrova una norma infinito che potrebbe essere un behaviour non desiderato